tasks/main.yml

```diff
@@ -328,13 +328,55 @@
     __mssql_conf_setting_value: "{{ mssql_ip_address }}"
   when: mssql_ip_address is not none
 
-- name: Configure the TCP port setting
+- name: Register the previous tcpport setting
+  when:
+    - mssql_firewall_configure | bool
+    - mssql_firewall_close_previous | bool
+  shell: grep 'tcpport' {{ __mssql_conf_path }} || true
```

Comment:
I thought there was a mssql-conf command to do this?  If not, then you might want to be more restrictive about the pattern used e.g. `grep '^tcpport[ ]' or something like that to avoid spurious matches.  Also, using `shell` isn't recommended - is there a way to use `command` and `failed_when` instead?

Reply:
Hi Rich,
I replaced `|| true` with `failed_when: false` and replaced `shell` with `command` because piping is now not necessary.
I made patters to be more restrictive like `'^tcpport = '` in all places.
I didn't use `/opt/mssql/bin/mssql-conf get` because it returns `No setting for the given option found in '/var/opt/mssql/mssql.conf'.` if tcpport setting is not set, even though the value is 1433 by default. And this happens inconsistently with different settings. And grepping the config file is simpler and cleaner.
I am planning to create a module for configuring mssql, it should resolve this issue in future. For now I think it's ok.